src/main/kotlin/com/cognifide/gradle/aem/common/file/transfer/FileTransferSmb.kt

```diff
@@ -0,0 +1,77 @@
+package com.cognifide.gradle.aem.common.file.transfer
+
+import com.cognifide.gradle.aem.common.AemException
+import com.cognifide.gradle.aem.common.file.FileException
+import com.cognifide.gradle.aem.common.file.IoTransfer
+import java.io.File
+import java.io.IOException
+import jcifs.smb.NtlmPasswordAuthentication
+import jcifs.smb.SmbException
+import jcifs.smb.SmbFile
+
+class FileTransferSmb(
+    uploadUrl: String,
+    private val credentials: Credentials,
+    private val domain: String = "",
+    private val ioTransfer: IoTransfer = IoTransfer()
+) : FileTransfer {
+
+    private val uploadUrl = if (uploadUrl.endsWith('/')) {
+        uploadUrl
+    } else {
+        "$uploadUrl/"
+    }
+
+    init {
+        if (!file(uploadUrl).isDirectory) {
+            throw AemException("aem.backup.uploadUrl must be a directory: '$uploadUrl'")
```

Comment:
same thing as for sftp / we cannot have direct correclation between backup <=> file transfer

Reply:
OK